Updated - Refactored ImageHandler request to be more re-usable, and checkboxes with missing `minimal` class for iCheck

snipe opened the pull request:
This just makes the UI a little more consistent on edit screens, so that the notes field is above the image delete/upload section, and all checkboxes use the correct CSS style (`minimal`, which is used by iCheck) and corrected the incorrect `aria-label`.

This also ended up fixing a few small bugs with uploads (avatars) on new user creation.

snipe committed:
Updated checkboxed with missing `minimal` class for iCheck

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated HTML for “standard” image edit/delete

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added arial labels

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use the imageHandler for profile avatar changes

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Refactor image handler

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added gravatar and avatar as fillable

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use fieldname for aria labels too

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Don’t offer the delete option if there is no avatar currently

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Corrected fieldname for user fieldname for avatar in Controller

Signed-off-by: snipe <snipe@snipe.net>

uberbrady commented on app/Http/Controllers/ProfileController.php:
```diff
@@ -67,37 +67,9 @@ public function postIndex(ImageUploadRequest $request)
         }
 
+        // Handle the avatar upload and/or delete if necessary
+        app('\App\Http\Requests\ImageUploadRequest')->handleImages($user, 600, 'avatar', 'avatars', 'avatar');
```
I've tried something like this before and miserably failed at being able to implement it. Very clever!

snipe replied:
It's not very well documented, but this is the only way I know of to use multiple form-requests within a certain controller method.

uberbrady commented on app/Http/Controllers/Users/UsersController.php:
```diff
@@ -131,7 +131,7 @@ public function store(SaveUserRequest $request)
 
         // we have to invoke the
-        app(\App\Http\Requests\ImageUploadRequest::class)->handleImages($user, 600, 'image', 'avatars', 'avatar');
+        app(\App\Http\Requests\ImageUploadRequest::class)->handleImages($user, 600, 'avatar', 'avatars', 'avatar');
```
Hrm. if you can do this this way, I wonder if you can do: `use \App\Http\Requests\ImageUploadRequest;` up top, then do just `app(ImageUploadRequest::class)` here? Really neither here nor there, I just prefer 'real' classnames to strings (also goes for the above usage in ProfileController too). But my (personal, not necessarily Grokability) preference for namespacing or `use` statements is if you use the class name more than once, you should do a `use` - if you're only using it once, then this format (the full namespaced class name) is fine.

snipe replied:
Done. Turns out we were already have `use App\Http\Requests\ImageUploadRequest` in in this file.

uberbrady commented on app/Http/Requests/ImageUploadRequest.php:
```diff
@@ -79,49 +81,39 @@ public function handleImages($item, $w = 600, $form_fieldname = null, $path = nu
             }
         }
 
-        if (is_null($form_fieldname)) {
```
I *love* putting this logic back up into the method definition, much less code :)

uberbrady commented on app/Http/Requests/ImageUploadRequest.php:
```diff
@@ -79,49 +81,39 @@ public function handleImages($item, $w = 600, $form_fieldname = null, $path = nu
 
                 $ext = $image->getClientOriginalExtension();
-                $file_name = $type.'-'.$form_fieldname.'-'.str_random(10).'.'.$ext;
+                $file_name = $type.'-'.$form_fieldname.'-'.$item->id.'-'.str_random(10).'.'.$ext;
```
Since we *store* filenames in the DB, this naming-scheme change shouldn't matter, right? I do like it - I love the idea that the ID of the thing will get reflected in the filename; could really be useful for tracking down bugs (not that we *EVER* have any, of course).

snipe replied:
Yeah, I had originally wanted it to include `thing-name`, but on create, we don't have thing-name yet. I think it will help with debugging tho.

uberbrady commented on app/Http/Requests/ImageUploadRequest.php:
```diff
@@ -146,30 +138,30 @@ public function handleImages($item, $w = 600, $form_fieldname = null, $path = nu
 
                  // Remove Current image if exists
-                if (($item->{$use_db_field}!='') && (Storage::disk('public')->exists($path.'/'.$item->{$use_db_field}))) {
+                if (($item->{$form_fieldname}!='') && (Storage::disk('public')->exists($path.'/'.$item->{$db_fieldname}))) {
```
Ooh! Sneaky fix :)

snipe replied:
I'm not even 100% sure if this is a good fix, but as I was refactoring (and filling my logs with debug code) I realized that those values are basically always the same.

uberbrady commented on app/Http/Requests/ImageUploadRequest.php:
```diff
@@ -146,30 +138,30 @@ public function handleImages($item, $w = 600, $form_fieldname = null, $path = nu
-                } catch (\Exception $e) {
-                    \Log::debug($e);
-                }
+        } elseif ($this->input('image_delete') == '1') {
```
Another excellent cleanup.

uberbrady reviewed (approved):
This looks like a great set of cleanups! I imagine you going through all of these screens and seeing everything that was wrong, and fixing it all at once. I think it's a nice collection of UI tweaks that will make the product look better. However, I think the name of the PR is pretty misleading - can you update it to include the new image handling and UI tweaks, in addition to the iCheckbox stuff? Regardless of that, that's not enough for me to block this. Thanks for fixing it and making the product nicer and nicer to work with!

snipe committed:
Shorten the additiona form request invocation to just `ImageUploadRequest`

We already have the `use App\Http\Requests\ImageUploadRequest` clause at the top of this file.

Signed-off-by: snipe <snipe@snipe.net>
